input/elasticapm/processor.go

```diff
@@ -296,19 +296,25 @@ func (p *Processor) handleStream(
 	if n == 0 {
 		return readErr
 	}
-
-	if err := p.processBatch(ctx, processor, batch); err != nil {
+	if err := processor.ProcessBatch(ctx, batch); err != nil {
 		return fmt.Errorf("cannot process batch: %w", err)
 	}
+	for _, v := range *batch {
+		switch v.Type() {
+		case modelpb.SpanEventType:
+			result.SpanAccepted++
+		case modelpb.TransactionEventType:
+			result.TransactionAccepted++
+		case modelpb.MetricEventType:
+			result.MetricAccepted++
+		case modelpb.LogEventType:
+			result.LogAccepted++
+		}
+	}
 	result.Accepted += n
 	return readErr
 }
 
-// processBatch processes the batch and returns the events to the pool after it's been processed.
-func (p *Processor) processBatch(ctx context.Context, processor modelpb.BatchProcessor, batch *modelpb.Batch) error {
-	return processor.ProcessBatch(ctx, batch)
-}
-
```

Comment:
**Reviewers**: I do not see any use for this method.

Reply:
I think it's a leftover from when we introduced async processing, but we never removed it.